src/kOS/Screen/GUIWindow.cs

```diff
@@ -0,0 +1,294 @@
+                }
+            }
+        }
+
+        void Update()
+        {
+            if (!IsPowered) {
+                Close();
+                Destroy(gameObject);
+                return;
+            }
+
+            if (shared == null || shared.Vessel == null || shared.Vessel.parts.Count == 0)
+            {
+                // Holding onto a vessel instance that no longer exists?
+                Close();
+            }
+
+            if (!IsOpen ) return;
+         
+            UpdateLogic();
+        }
+
+        void WidgetGui(int windowId)
+        {
+            if (widgets == null) return;
+
+            CheckConnectivity();
+
+            widgets.DoGUI();
+            if (delayedActions.Count > 0) {
+                var c = GUI.color;
+                GUI.color = new Color(1,1,1, Mathf.PingPong(Time.realtimeSinceStartup * 0.9f, 0.2f)+0.4f);
+                var delay = delayedActions[0].time - shared.UpdateHandler.CurrentTime;
+                var textHeight = 30;
+                var rect = new Rect((WindowRect.width-commDelayedTexture.width) / 2, (WindowRect.height-commDelayedTexture.height- textHeight) / 2, commDelayedTexture.width, commDelayedTexture.height);
+                GUI.DrawTexture(rect, commDelayedTexture);
+                rect.y += commDelayedTexture.height;
+                rect.x -= 10;
+                rect.width += 20;
+                rect.height = textHeight;
+                GUI.Label(rect, string.Format("{1}\n{0:0.0} seconds", delay,delayedActions[0].reason), commDelayStyle);
+                GUI.color = c;
+            }
+
+            if (draggable)
+                GUI.DragWindow();
+        }
+
+        void CheckConnectivity()
+        {
+            if (!ConnectivityManager.HasConnectionToControl(shared.Vessel)) {
+                GUI.color = noControlColor;
+                GUI.enabled = false;
+            } else {
+                var mdelay = ConnectivityManager.GetDelayToControl(shared.Vessel) + extraDelay;
+                if (mdelay > 0.1)
+                    GUI.color = slowControlColor;
+            }
+        }
+
+        void PopupGui(int windowId)
+        {
+            if (currentPopup != null) {
+                CheckConnectivity();
+                currentPopup.DoPopupGUI();
+            }
+        }
+        
+        public Rect GetRect()
+        {
+            return WindowRect;
+        }
+
+        public void SetX(float x)
+        {
+            var r = WindowRect;
+            if (x < 0) r.x = UnityEngine.Screen.width + x - r.width + 1; // X11-geometry-style negative position
+            else r.x = x;
+            WindowRect = r;
+        }
+
+        public void SetY(float y)
+        {
+            var r = WindowRect;
+            if (y < 0) r.y = UnityEngine.Screen.height + y - r.height + 1;
+            else r.y = y;
+            WindowRect = r;
+        }
+
+        public PopupMenu currentPopup;
+
```

Comment:
While not "gospel", the referenence guidlines state:
>Classes member should be alphabetized, and grouped into sections (Fields, Constructors, Properties, Events, Methods, Private interface implementations, Nested types)

I'm not one to get too hung up on it, cause it isn't a huge issue (particularly the alphabetizing).  But if most of the fields/properties are declared at the top, it does make it easier to read the file if that location is consistent.  I would probably move them if I were doing the review.